fastapi/schemas.py

```diff
@@ -4,21 +4,20 @@
 from enum import Enum
 from typing import Generic, List, Optional, TypeVar
 
-from pydantic import BaseModel, Field
-from pydantic.generics import GenericModel
+from pydantic import BaseModel, ConfigDict, Field
 
 T = TypeVar("T")
 
 
-class PagedCollection(GenericModel, Generic[T]):
+class PagedCollection(BaseModel, Generic[T]):
 
     total: int
     items: List[T]
 
 
 class Paging(BaseModel):
-    limit: Optional[int]
-    offset: Optional[int]
+    limit: Optional[int] = None
+    offset: Optional[int] = None
```

Comment:
According to Pydantic migration guide:

_Pydantic V2 changes some of the logic for specifying whether a field annotated as Optional is required (i.e., has no default value) or not (i.e., has a default value of None or any other value of the corresponding type)._

Hence we must distinguish:
* `Optional[str]` (Required, can be `None`), and
* `Optional[str] = None` (Not required, can be `None`, is `None` by default).

Are we sure we want to make this update here?

See https://docs.pydantic.dev/latest/migration/#:~:text=Pydantic%20V2%20changes,annotations%20in%20V2%3A